src/test/java/com/snowflake/kafka/connector/internal/StageFilesProcessorTest.java

```diff
@@ -0,0 +1,249 @@
+package com.snowflake.kafka.connector.internal;
+
+import com.snowflake.kafka.connector.internal.telemetry.SnowflakeTelemetryPipeCreation;
+import com.snowflake.kafka.connector.internal.telemetry.SnowflakeTelemetryPipeStatus;
+import com.snowflake.kafka.connector.internal.telemetry.SnowflakeTelemetryService;
+import net.snowflake.client.jdbc.internal.joda.time.DateTime;
+import net.snowflake.client.jdbc.internal.joda.time.DateTimeZone;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mockito;
+
+import java.time.Duration;
+import java.util.*;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.*;
+
+
+class StageFilesProcessorTest {
+
+    private static final String PIPE_NAME = "testPipe";
+    private static final String TABLE_NAME = "testTable";
+    private static final String STAGE_NAME = "testStage";
+    private static final String PREFIX = "prefix";
+
+    private SnowflakeConnectionService conn;
+    private SnowflakeIngestionService ingestionService;
+    private AtomicLong currentTime;
+    private StageFilesProcessor victim;
+    private StageFilesProcessor.ProgressRegisterImpl register;
+    private SnowflakeTelemetryPipeCreation pipeCreation;
+    private SnowflakeTelemetryPipeStatus pipeTelemetry;
+    private SnowflakeTelemetryService telemetryService;
+
+    @BeforeEach
+    void setUp() {
+        currentTime = new AtomicLong(new DateTime(2000, 1, 10, 12, 0, DateTimeZone.UTC).getMillis());
+
+        conn = Mockito.mock(SnowflakeConnectionService.class);
+        ingestionService = Mockito.mock(SnowflakeIngestionService.class);
+        pipeCreation = new SnowflakeTelemetryPipeCreation(TABLE_NAME, STAGE_NAME, PIPE_NAME);
+        pipeTelemetry = Mockito.mock(SnowflakeTelemetryPipeStatus.class);
+        telemetryService = Mockito.mock( SnowflakeTelemetryService.class);
+        register = new StageFilesProcessor.ProgressRegisterImpl(pipeCreation, pipeTelemetry, telemetryService);
+
+        victim = new StageFilesProcessor(PIPE_NAME, TABLE_NAME, STAGE_NAME, PREFIX, conn, ingestionService, pipeTelemetry, telemetryService, currentTime::get);
```

Comment:
I haven't seen anyone naming SUTs victims :)
It wasn't really obvious from the tests what is represented by it.

Reply:
old habit :)